File: py/tests/testSearchInRotatedSortedArr/test_SearchInRotatedSortedArr.py

import unittest
from parameterized import parameterized as p
from solns.searchInRotatedSortedArr.searchInRotatedSortedArr import *

class UnitTest_SearchInRotatedSortedArr(unittest.TestCase):
    @p.expand([
        [[4,5,6,7,0,1,2],0,4],[[4,5,6,7,0,1,2],3,-1],[[1],0,-1]
    ])
    def test_naive(self,nums,target,expected):
        self.assertEqual(Solution.naive(nums,target),expected)

File: cs/sample/Sample.Tests/SearchInRotatedSortedArr/testSearchInRotatedSortedArr.cs
using System;
using Xunit;
using Sample.SearchInRotatedSortedArr;

namespace Sample.Tests{
    public class UnitTest_SearchInRotatedSortedArr{
        [Theory]
        [InlineData(new int[]{4,5,6,7,0,1,2},0,4)]
        [InlineData(new int[]{4,5,6,7,0,1,2},3,-1)]
        [InlineData(new int[]{1},0,-1)]
        public void checkResult(int[] nums,int target,int expected){
            Assert.Equal(expected,searchInRotatedSortedArr.naive(nums,target));
        }
    }
}
